lib/spack/spack/schema/modules.py

```diff
@@ -17,7 +17,7 @@
 #: THIS NEEDS TO BE UPDATED FOR EVERY NEW KEYWORD THAT
 #: IS ADDED IMMEDIATELY BELOW THE MODULE TYPE ATTRIBUTE
 spec_regex = (
-    r"(?!hierarchy|core_specs|verbose|hash_length|defaults|"
+    r"(?!hierarchy|core_specs|verbose|hash_length|defaults|filter_hierarchy_specs"
```

Comment:
@becker33 you need a `|` at the end of this to resolve the test failures

Reply:
The number of times my eyes have just glided over that without noticing... thank you!